Gordon360/Services/NewsService.cs

```diff
@@ -240,21 +240,45 @@ public StudentNewsViewModel EditPosting(int newsID, StudentNews newData)
 
             if (newData.Image != null)
             {
-                string fileName = newsItem.SNID + ".jpg";
-                string imagePath = NewsUploadsPath + fileName;
-                ImageUtils.UploadImage(imagePath, newData.Image);
-                newsItem.Image = imagePath;
+                // ImageUtils.GetImageFormat checks whether the image type is valid (jpg/jpeg/png)
+                var (extension, format, data) = ImageUtils.GetImageFormat(newData.Image);
+
+                // If there's already an image attached to the news post, simply overwrite the old image
+                // data when uploading a new one
+                if (newsItem.Image != null)
+                {
+                    var imagePath = Path.Combine(_webHostEnvironment.ContentRootPath, "browseable", "uploads", Path.GetFileName(newsItem.Image));
+                    ImageUtils.UploadImage(imagePath, newData.Image.Split(",").Last());
+                }
+
+                // If there isn't an image attached to the news post, do the same procedure to save
+                // image in Submit method
+                else
+                {
+                    // Use a unique alphanumeric GUID string as the file name
+                    var filename = $"{Guid.NewGuid().ToString("N")}.{extension}";
+                    var imagePath = Path.Combine(_webHostEnvironment.ContentRootPath, "browseable", "uploads", filename);
+
+                    var serverAddress = _serverUtils.GetAddress();
+                    if (serverAddress is not string) throw new Exception("Could not upload Student News Image: Server Address is null");
+
+                    var url = $"{serverAddress}/browseable/uploads/{filename}";
+
+                    ImageUtils.UploadImage(imagePath, newData.Image.Split(",").Last());
+
+                    newsItem.Image = url;
+                }
```

Comment:
I think you could restructure this to be a little bit less redundant in the if and else branches here.

Also, please pass `format` to `UploadImage`.

```suggestion
                // ImageUtils.GetImageFormat checks whether the image type is valid (jpg/jpeg/png)
                var (extension, format, data) = ImageUtils.GetImageFormat(newData.Image);
                
                string? imagePath = null

                // If old image exists, overwrite it with new image at same path
                if (newsItem.Image != null)
                {
                    imagePath = Path.Combine(_webHostEnvironment.ContentRootPath, "browseable", "uploads", Path.GetFileName(newsItem.Image));
                }

                // Otherwise, upload new image and save url to db
                else
                {
                    // Use a unique alphanumeric GUID string as the file name
                    var filename = $"{Guid.NewGuid().ToString("N")}.{extension}";
                   imagePath = Path.Combine(_webHostEnvironment.ContentRootPath, "browseable", "uploads", filename);

                    var serverAddress = _serverUtils.GetAddress();
                    if (serverAddress is not string) throw new Exception("Could not upload Student News Image: Server Address is null");

                    var url = $"{serverAddress}/browseable/uploads/{filename}";

                    newsItem.Image = url;
                }
                var imageData = newData.Image.Split(",").Last();
                
                ImageUtils.UploadImage(imagePath, imageData, format);
```